src/main/java/marquez/common/models/Description.java

```diff
@@ -14,18 +14,18 @@
 
 package marquez.common.models;
 
+import javax.annotation.Nullable;
+import lombok.AllArgsConstructor;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
-import lombok.NonNull;
-import lombok.RequiredArgsConstructor;
 import lombok.ToString;
 
-@RequiredArgsConstructor(staticName = "fromString")
+@AllArgsConstructor(staticName = "fromString")
 @EqualsAndHashCode
 @ToString
 public final class Description {
-  public static final String NO_VALUE = "";
+  public static final String NO_VALUE = null;
   public static final Description NO_DESCRIPTION = new Description(NO_VALUE);
```

Comment:
Hm, not sure I understand `NO_DESCRIPTION`.. it's less clear what's happening elsewhere in the code. `new Description(NO_VALUE)` is almost easier to understand

Reply:
If we're going to avoid using `null` to signify that something is not set, then I think that value should live outside of `marquez.common.models.Description` and should be used universally in place of `null`. I haven't worked with a ton of java codebases, but to me it feels very natural to use `null` when something is not set. Is this considered bad form?

Reply:
> Hm, not sure I understand NO_DESCRIPTION.. it's less clear what's happening elsewhere in the code. new Description(NO_VALUE) is almost easier to understand

The static instance `Description.NO_DESCRIPTION` is more for convenience and expressibility. It's a type that may or may not be present (`null`, empty / blank string), so we want to handle the optional type differently, or at least attempt to avoid the occurrences of `new Description(null)` throughout our codebase.

For example, let's look at an example `Row` class:

```java
public final class Row {
  @Getter private final UUID uuid;
  @Getter private final Instant createdAt;
  @Nullable private final Description description;

  public Optional<String> getDescription() {
    return Optional.ofNullable(description);
  }
}
```

It's clear `Row.Description` maybe `null`. Let's say we want to read the `DatasetRow.Description`. We can either:

```java
// Option 1
final Row row = ...
final Description description = row.getDescription().orElse(new Description(null))
```

or use a statically defined instance given the common occurrence of attempting the access a row description

```java
import static marquez.common.models.Description.NO_DESCRIPTION;

// Option 1
final Row row = ...
final Description description = row.getDescription().orElse(NO_DESCRIPTION)
```

> If we're going to avoid using null to signify that something is not set, then I think that value should live outside of marquez.common.models.Description

I don't think `null`s in a codebase is a bad thing, and in most cases unavoidable. To begin replacing all `null`s with `NO_VALUE` seems a bit much and am ok with removing the constant. That is, 

```java
public final class Description {
  ...
  public static final Description NO_DESCRIPTION = new Description(null);
}
```

Reply:
Gotcha.
I'd be ok with keeping `NO_DESCRIPTION`, but would rather stick with `null`s over `NO_VALUE`s in the code.